rotkehlchen/inquirer.py

```diff
@@ -388,8 +440,84 @@ def find_usd_price(
+                price = Price(ZERO)
+            else:
+                price = Price(usd_price)
+            Inquirer._cached_current_price[cache_key] = CachedPriceEntry(
+                price=price,
+                time=ts_now(),
+            )
+            return price
+
         return instance._query_oracle_instances(from_asset=asset, to_asset=A_USD)
 
+    def find_uniswap_v2_lp_price(
+        self,
+        token: Union[EthereumToken, UnknownEthereumToken],
+    ) -> Optional[Price]:
+        """
+        First thing should be query oracles for prices. It it files then we
+        calculate price for a LP token. This one is
+        value = (Total value of liquidity pool)/(Current suply of LP tokens)
+        We need:
+        - Price of token 0
+        - Price of token 1
+        - Pooled amount of token 0
+        - Pooled amount of token 1
+        - Total suply of of pool token
+        """
+        address = token.ethereum_address
+        contract = EthereumContract(address=address, abi=UNISWAP_V2_LP_ABI, deployed_block=0)
+        methods = ['token0', 'token1', 'totalSupply', 'getReserves', 'decimals']
+
+        if self._ethereum is None:
+            log.debug(
+                'Tried to query uniswap v2 LP position price but '
+                'Inquierer is not correctly initializen',
+            )
+            return None
+
+        try:
+            output = multicall_2(
+                ethereum=self._ethereum,
+                require_success=True,
+                calls=[(address, contract.encode(method_name=method)) for method in methods],
+            )
+        except RemoteError:
+            log.debug('Remote error calling multicall contract for uniswap v2 lp token properties')
+            return None
+
+        # decode output
+        decoded = []
+        for (method_output, method_name) in zip(output, methods):
+            if method_output[0] and len(method_output[1]) != 0:
+                decoded_method = contract.decode(method_output[1], method_name)
+                if len(decoded_method) == 1:
+                    decoded.append(decoded_method[0])
+                else:
+                    decoded.append(decoded_method)
+            else:
+                log.debug(f'Multicall to Uniswap V2 LP failed to fetch field {method_name}')
+                return None
+
+        try:
+            token0 = EthereumToken(decoded[0])
+            token1 = EthereumToken(decoded[1])
+        except UnknownAsset:
+            return None
+        token0_price = self.find_usd_price(token0)
+        token1_price = self.find_usd_price(token1)
+        if ZERO not in (token0_price, token1_price):
+            token0_supply = FVal(decoded[3][0] * 10**-token0.decimals)
+            token1_supply = FVal(decoded[3][1] * 10**-token1.decimals)
+
+        numerator = (token0_supply * token0_price + token1_supply * token1_price)
+        share_value = numerator / FVal(decoded[2] * 10 ** - decoded[4])
```

Comment:
FVal initialization can fail and the ValueError should be handled hre.

Also please initialize all participating numbers as FVal and THEN do the calculations if they are supposed to be floating point numbers, as they are more accurate.

Can't speak as to the calculation itself. But I guess its' something easily testable.